src/main/scala/vectorpipe/functions/osm/package.scala

```diff
@@ -0,0 +1,227 @@
+      "embankment" -> true,
+      "groyne" -> true,
+      "pier" -> true,
+      "pipeline" -> true
+    ),
+    "military" -> Map(),
+    "natural" -> Map(
+      "cliff" -> true,
+      "coastline" -> true,
+      "ridge" -> true,
+      "tree_row" -> true
+    ),
+    "office" -> Map(),
+    "piste:type" -> Map(),
+    "place" -> Map(),
+    "playground" -> Map(
+      "balancebeam" -> true,
+      "slide" -> true,
+      "zipwire" -> true
+    ),
+    "power" -> Map(
+      "line" -> true,
+      "minor_line" -> true
+    ),
+    "public_transport" -> Map(
+      "platform" -> true
+    ),
+    "shop" -> Map(),
+    "tourism" -> Map(
+      "artwork" -> true
+    ),
+    "waterway" -> Map(
+      "canal" -> true,
+      "dam" -> true,
+      "ditch" -> true,
+      "drain" -> true,
+      "river" -> true,
+      "stream" -> true,
+      "weir" -> true
+    )
+  )
+
+  private val MultiPolygonTypes = Seq("multipolygon", "boundary")
+
+  private val BooleanValues = Seq("yes", "no", "true", "false", "1", "0")
+
+  private val TruthyValues = Seq("yes", "true", "1")
+
+  private val WaterwayValues =
+    Seq(
+      "river", "canal", "stream", "brook", "drain", "ditch"
+    )
+
+  private val POITags = Set("amenity", "shop", "craft", "office", "leisure", "aeroway")
+
+  private val HashtagMatcher: Regex = """#([^\u2000-\u206F\u2E00-\u2E7F\s\\'!\"#$%()*,.\/;<=>?@\[\]^{|}~]+)""".r
+
+  private val _isArea = (tags: Map[String, String]) =>
+    tags match {
+      case _ if tags.contains("area") && BooleanValues.contains(tags("area").toLowerCase) =>
+        TruthyValues.contains(tags("area").toLowerCase)
+      case _ =>
+        // see https://github.com/osmlab/id-area-keys (values are inverted)
+        val matchingKeys = tags.keySet.intersect(AreaKeys.keySet)
+        matchingKeys.exists(k => !AreaKeys(k).contains(tags(k)))
+    }
+
+  val isAreaUDF: UserDefinedFunction = udf(_isArea)
+
+  def isArea(tags: Column): Column =
+    when(lower(coalesce(tags.getField("area"), lit(""))).isin(BooleanValues: _*),
+         lower(tags.getField("area")).isin(TruthyValues: _*))
+      // only call the UDF when necessary
+      .otherwise(isAreaUDF(tags)) as 'isArea
+
+  def isMultiPolygon(tags: Column): Column =
+    lower(coalesce(tags.getItem("type"), lit("")))
+      .isin(MultiPolygonTypes: _*) as 'isMultiPolygon
+
+  def isNew(version: Column, minorVersion: Column): Column =
+    version <=> 1 && minorVersion <=> 0 as 'isNew
+
+  def isRoute(tags: Column): Column =
+    lower(tags.getItem("type")) <=> "route" as 'isRoute
+
+  private lazy val MemberSchema = ArrayType(
+    StructType(
+      StructField("type", ByteType, nullable = false) ::
+        StructField("ref", LongType, nullable = false) ::
+        StructField("role", StringType, nullable = false) ::
+        Nil), containsNull = false)
+
+  private val _compressMemberTypes = (members: Seq[Row]) =>
+    members.map { row =>
+      val t = row.getAs[String]("type") match {
```

Comment:
I added a static function to Member to do this conversion.  Maybe we should rely on that more broadly?

Reply:
Oh yes. I feel like the mapping from String ↔︎ Byte exists in a number of places.